### PR TITLE
add v0.4.0 changelog

### PR DESCRIPTION
**Pull request by jiangpengcheng**
### Motivation

Add v0.4.0 changelog

### Modifications

Add v0.4.0 changelog

### Verifying this change

- [x] Make sure that the change passes the CI checks.

- [x] This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below.

Need to update docs? 

- [x] `no-need-doc` 
  
  release

### CODE REVIEW

**Review by tpiperatgod — APPROVED** [review]

**Review by freeznet — APPROVED** [review]
@Huanli-Meng could you PTAL when you have time, thanks.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,25 @@
+# Function Mesh v0.4.0 Release Notes
+
+## v0.4.0 What's New
+
+- [Makefile] Add `kustomize` before `crd` in `release` command in Makefile ([#395](https://github.com/streamnative/function-mesh/pull/395))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:5]:
```suggestion
- [Makefile] Add `kustomize` before `crd` in the `release` command in Makefile ([#395](https://github.com/streamnative/function-mesh/pull/395))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+- [Makefile] remove `helm-crds` from `release` command in Makefile ([#396](https://github.com/streamnative/function-mesh/pull/396))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:6]:
```suggestion
- [Makefile] Remove `helm-crds` from the `release` command in Makefile ([#396](https://github.com/streamnative/function-mesh/pull/396))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+- [Makefile] Some small types fixed in readme and makefile ([#423](https://github.com/streamnative/function-mesh/pull/423))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:7]:
```suggestion
- [Makefile] Fix some small typos in README and Makefile ([#423](https://github.com/streamnative/function-mesh/pull/423))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,25 @@
...
+- [Test] wait for controller POD ready in Github Action ([#406](https://github.com/streamnative/function-mesh/pull/406))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:8]:
```suggestion
- [Test] Fix failed actions to wait for the controller POD ready in Github Action ([#406](https://github.com/streamnative/function-mesh/pull/406))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,25 @@
...
+- [Test] add `paths-ignore` configuration ([#410](https://github.com/streamnative/function-mesh/pull/410))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:9]:
```suggestion
- [Test] Add the `paths-ignore` configuration to allow changing some files without triggering the CI test workflow. ([#410](https://github.com/streamnative/function-mesh/pull/410))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,25 @@
...
+- [Test] update CI workflow ([#420](https://github.com/streamnative/function-mesh/pull/420))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:10]:
```suggestion
- [Test] Update the CI test workflow ([#420](https://github.com/streamnative/function-mesh/pull/420))
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,25 @@
...
+- [Test] Add `make e2e` command for e2e tests based on skywalking-e2e-infra ([#422](https://github.com/streamnative/function-mesh/pull/422))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:11]:
```suggestion
- [Test] Add the `make e2e` command for E2E tests based on [SkyWalking Infra E2E](https://github.com/apache/skywalking-infra-e2e) ([#422](https://github.com/streamnative/function-mesh/pull/422))
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,25 @@
...
+- [Helm Charts] Remove invalid caBundle in webhook patches ([#398](https://github.com/streamnative/function-mesh/pull/398))
+- [Helm Charts] Improve tools for generating helm chart templates ([#409](https://github.com/streamnative/function-mesh/pull/409))
+- [Controller] update the codeowners ([#407](https://github.com/streamnative/function-mesh/pull/407))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:14]:
```suggestion
- [Controller] Update the codeowners ([#407](https://github.com/streamnative/function-mesh/pull/407))
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,25 @@
...
+- [Controller] Fix dockerfile comment ([#405](https://github.com/streamnative/function-mesh/pull/405))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:15]:
```suggestion
- [Controller] Fix comments on the Dockerfile ([#405](https://github.com/streamnative/function-mesh/pull/405))
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,25 @@
...
+- [Controller] Remove useless omitempty ([#402](https://github.com/streamnative/function-mesh/pull/402))
+- [Controller] Use secret for trust path ([#414](https://github.com/streamnative/function-mesh/pull/414))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:17]:
```suggestion
- [Controller] Support mounting secrets to provide sensitive information to Function Mesh ([#414](https://github.com/streamnative/function-mesh/pull/414))
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,25 @@
...
+- [Controller] Allow key-value Environment variable passthrough. ([#424](https://github.com/streamnative/function-mesh/pull/424))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:18]:
```suggestion
- [Controller] Allow key-value environment variable passthrough ([#424](https://github.com/streamnative/function-mesh/pull/424))
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,25 @@
...
+- [Controller] use ctrl.CreateOrUpdate to apply resource ([#427](https://github.com/streamnative/function-mesh/pull/427))
```
Comment (Huanli-Meng) [inline CHANGELOG.md:19]:
```suggestion
- [Controller] Use `ctrl.CreateOrUpdate` to determine whether a resource should be updated or created ([#427](https://github.com/streamnative/function-mesh/pull/427))
```

**Review by Huanli-Meng — COMMENTED** [review]
Leave my comments. PTAL

**Comment by jiangpengcheng** [comment]
> Leave my comments. PTAL

thanks a lot for the suggestions, already applied

**Review by Huanli-Meng — APPROVED** [review]
LGTM

**Pull request merged**